lib/createJestRunner.ts

```diff
@@ -1,20 +1,41 @@
-import throat from 'throat';
+import type * as JestResult from '@jest/test-result';
+import type { Config } from '@jest/types';
+import type * as JestRunner from 'jest-runner';
 import Worker from 'jest-worker';
+import throat from 'throat';
+import type { Path, TestRunner } from './types';
 
 class CancelRun extends Error {
-  constructor(message) {
+  constructor(message?: string) {
     super(message);
     this.name = 'CancelRun';
   }
 }
 
-const createRunner = (runPath, { getExtraOptions } = {}) => {
-  class BaseTestRunner {
-    constructor(globalConfig) {
+export default function createRunner<ExtraOptionsType extends object>(
+  runPath: Path,
+  { getExtraOptions }: { getExtraOptions?: () => ExtraOptionsType } = {},
+): typeof TestRunner {
+  return class BaseTestRunner implements TestRunner {
+    declare readonly _globalConfig: Config.GlobalConfig;
```

Comment:
We should enable class properties babel transform. https://github.com/facebook/jest/blob/0008d782db653b13da2656e5f4f5dd874275deea/babel.config.js#L58

Reply:
Looks like the yarn.lock has conflicts, should I rebase my commits on top of the current master?

Reply:
Yep 👍

Reply:
I squashed some of the commits (updating babelrc, tsconfig) when I rebased to keep the commit history cleaner.  
Kept the co-author commit footer in some of the commits you helped reviewing, I hope that's ok with you.